tests/OSPSuite.Core.IntegrationTests/DiffBuilders/ContainerObjectComparerSpecs.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Linq;
+using OSPSuite.BDDHelper;
+using OSPSuite.BDDHelper.Extensions;
+using OSPSuite.Core.Domain;
+
+namespace OSPSuite.Core.DiffBuilders
+{
+   public class When_comparing_two_container_having_different_base_properties_and_container_properties : concern_for_ObjectComparer
```

Comment:
Why did you rename the file. We are testing the behavior of ContainerDiff. We do not have an object called ContainerObjectComparer

Reply:
you are right about that, but this was called "ContainterDiffBuilderSpecs", we do have a ContainterDiffBuilder class abd this tests are testing the "ContainerDiffBuilder" at all.

Since the concern class is concern_for_ObjectComparer : ContextForIntegration<IObjectComparer> , I thought it should be named after that.
Let me know what do you think

Reply:
this is not correct. We are testing the behavior of the ContainerDiff Builder implictely by making the diff of two contaienrs. This is how it goes here. Instead of always duplicating the logic, we created one base class, and we always create in the context the two objects to compare. I want to test the ContainerDiffBuilder, I create two containers  etc..

Because it's integration tests, we are testing the whole system as a whole but we are focusing here on Container. So in our case, we ARE Testing container diff builder by creating TWO containers. You just need to update the context to have two different ParentPath and that's it